uniffi_bindgen/src/bindings/kotlin/templates/RecordTemplate.kt

```diff
@@ -33,11 +33,11 @@ data class {{ rec.name()|class_name_kt }} (
         {% endfor %}
     }
 
-    {% if rec.contains_object_references(ci) %}
+    {% if ci.contains_object_references(rec) %}
     @Suppress("UNNECESSARY_SAFE_CALL") // codegen is much simpler if we unconditionally emit safe calls here
     override fun destroy() {
         {% for field in rec.fields() %}
-            {%- if ci.type_contains_object_references(field.type_()) -%}
+            {%- if ci.contains_object_references(field) -%}
```

Comment:
OTOH, maybe the name here needs some sort of subject indicator as the first word...`ci.item_contains_object_references` or similar?